---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r, include = FALSE}
knitr::opts_chunk$set(
    collapse = TRUE,
    comment = "#>",
    fig.path = "man/figures/README-",
    out.width = "100%"
)
```

# spoar

<!-- badges: start -->
[![Lifecycle: experimental](https://img.shields.io/badge/lifecycle-experimental-orange.svg)](https://lifecycle.r-lib.org/articles/stages.html#experimental)
[![BioC status](http://www.bioconductor.org/shields/build/release/bioc/spoar.svg)](https://bioconductor.org/checkResults/release/bioc-LATEST/spoar)
[![R-CMD-check-bioc](https://github.com/brendanf/spoar/workflows/R-CMD-check-bioc/badge.svg)](https://github.com/brendanf/spoar/actions)
[![Codecov test coverage](https://codecov.io/gh/brendanf/spoar/branch/main/graph/badge.svg)](https://codecov.io/gh/brendanf/spoar?branch=main)
<!-- badges: end -->

[SPOA](https://github.com/rvaser/spoa) is a C++ library for fast alignment and
consensus generation of closely related sequences, especially used for long,
error-containing reads from third generation sequencers. It allows local,
global, and semi-global (overlap) alignment, using linear, affine, or convex
gap penalties, and can also weight by quality scores. SPOAR is an R wrapper for
SPOA.

## Installation instructions

Get the latest stable `R` release from [CRAN](http://cran.r-project.org/). Then install `spoar` using from [Bioconductor](http://bioconductor.org/) the following code:

```{r 'install', eval = FALSE}
if (!requireNamespace("BiocManager", quietly = TRUE)) {
    install.packages("BiocManager")
}

BiocManager::install("spoar")
```

And the development version from [GitHub](https://github.com/brendanf/spoar) with:

```{r 'install_dev', eval = FALSE}
BiocManager::install("brendanf/spoar")
```
## Example

Here is the first basic example from the SPOR README:

```{r example, eval = requireNamespace('spoar')}
library("spoar")

sequences <- c(
    "CATAAAAGAACGTAGGTCGCCCGTCCGTAACCTGTCGGATCACCGGAAAGGACCCGTAAAGTGATAATGAT",
    "ATAAAGGCAGTCGCTCTGTAAGCTGTCGATTCACCGGAAAGATGGCGTTACCACGTAAAGTGATAATGATTAT",
    "ATCAAAGAACGTGTAGCCTGTCCGTAATCTAGCGCATTTCACACGAGACCCGCGTAATGGG",
    "CGTAAATAGGTAATGATTATCATTACATATCACAACTAGGGCCGTATTAATCATGATATCATCA",
    "GTCGCTAGAGGCATCGTGAGTCGCTTCCGTACCGCAAGGATGACGAGTCACTTAAAGTGATAAT",
    "CCGTAACCTTCATCGGATCACCGGAAAGGACCCGTAAATAGACCTGATTATCATCTACAT"
)

spoaAlign(sequences)

spoaConsensus(sequences)
```

## Citation

Below is the citation output from using `citation('spoar')` in R. Please
run this yourself to check for any updates on how to cite __spoar__.

```{r 'citation', eval = requireNamespace('spoar')}
print(citation("spoar"), bibtex = TRUE)
```

Please note that the `spoar` was only made possible thanks to many other R and bioinformatics software authors, which are cited either in the vignettes and/or the paper(s) describing this package.

## Code of Conduct

Please note that the `spoar` project is released with a [Contributor Code of Conduct](http://bioconductor.org/about/code-of-conduct/). By contributing to this project, you agree to abide by its terms.

## Development tools

* Continuous code testing is possible thanks to [GitHub actions](https://www.tidyverse.org/blog/2020/04/usethis-1-6-0/)  through `r BiocStyle::CRANpkg('usethis')`, `r BiocStyle::CRANpkg('remotes')`, and `r BiocStyle::CRANpkg('rcmdcheck')` customized to use [Bioconductor's docker containers](https://www.bioconductor.org/help/docker/) and `r BiocStyle::Biocpkg('BiocCheck')`.
* Code coverage assessment is possible thanks to [codecov](https://codecov.io/gh) and `r BiocStyle::CRANpkg('covr')`.
* The [documentation website](http://brendanf.github.io/spoar) is automatically updated thanks to `r BiocStyle::CRANpkg('pkgdown')`.
* The code is styled automatically thanks to `r BiocStyle::CRANpkg('styler')`.
* The documentation is formatted thanks to `r BiocStyle::CRANpkg('devtools')` and `r BiocStyle::CRANpkg('roxygen2')`.

For more details, check the `dev` directory.

This package was developed using `r BiocStyle::Biocpkg('biocthis')`.


